massa-api/src/public.rs

```diff
@@ -1155,3 +1117,76 @@ impl MassaRpcServer for API<Public> {
         openrpc
     }
 }
+
+/// Checks the validity of an input operation.
+///
+/// This function takes an `OperationInput`, an `APIConfig`, and an optional `Slot` as input parameters.
+/// It performs various checks on the input operation and returns a `SecureShareOperation` if the checks pass.
+/// Otherwise, it returns an `RpcResult` with an appropriate error.
+///
+/// # Arguments
+///
+/// * `op_input` - The input operation to be checked.
+/// * `api_cfg` - The API configuration used for checking the operation.
+/// * `last_slot` - An optional `Slot` representing the last slot used.
+///
+/// # Returns
+///
+/// Returns a `RpcResult` containing a `SecureShareOperation` if the input operation is valid.
+/// Otherwise, returns an `RpcResult` with an appropriate error.
+fn check_input_operation(
+    op_input: OperationInput,
+    api_cfg: &APIConfig,
+    last_slot: Option<Slot>,
+) -> RpcResult<SecureShareOperation> {
+    let operation_deserializer = SecureShareDeserializer::new(OperationDeserializer::new(
+        api_cfg.max_datastore_value_length,
+        api_cfg.max_function_name_length,
+        api_cfg.max_parameter_size,
+        api_cfg.max_op_datastore_entry_count,
+        api_cfg.max_op_datastore_key_length,
+        api_cfg.max_op_datastore_value_length,
+    ));
+
+    let mut op_serialized = Vec::new();
+    op_serialized.extend(op_input.signature.to_bytes());
+    op_serialized.extend(op_input.creator_public_key.to_bytes());
+    op_serialized.extend(op_input.serialized_content);
```

Comment:
Haven't checked what is available, but can't you just serialize the whole OperationInput directly?